### DISCUSSION TITLE
Project plan

### DISCUSSION THREAD

**Discussion by aayushchugh** [keizerworks/invoicen#10] [Announcements]
We'll start with MVP and let users create invoices without logging in directly through the front end. Users will have to add all the details every time in this MVP manually.

Fields that we'll cover in the invoice are mentioned in the Readme file of this repo.

## Invoice template inspiration:

![image 4](https://github.com/user-attachments/assets/6ab144f3-21b1-4107-be30-55c138b3e089)
![image 3](https://github.com/user-attachments/assets/582416c7-240b-49bc-8baf-281870cbbe0c)
![image 2](https://github.com/user-attachments/assets/0092f569-a3a3-4742-8f24-d58e080abcd7)
![image](https://github.com/user-attachments/assets/cc443f55-5e83-409c-85c0-82255a85016a)

**Comment by divyesh-mali** [comment]
Its really awesome 👏

**Comment by Prithviraj2003** [comment]
**Invoice Details :** 

1. Invoice Number): Input field (text)
2. Bill Date: Input field (date)
3. Delivery Date: Input field (date)
4. Terms of Payment: Input field (text, e.g., "Within 15 days")
5. Payment Deadline: Input field (date)

**Seller Details (Your Company Information) :**

1. Company Name: Input field (text)
2. Address: Input field (textarea)
3. Phone Number: Input field (text)
4. Email: Input field (email)
5. SIRET: Input field (text)
6. GSTIN: Input field (text)

**Billing Address :**

1. Customer Name: Input field (text)
2. Address: Input field (textarea)
3. Phone Number: Input field (text)
4. Email: Input field (email)
5. SIRET: Input field (text)
6. GSTIN: Input field (text)

**Items (Dynamic List) :**

1. Article: Input field (text)
2. Quantity: Input field (number)
3. Unit Price: Input field (number)
4. TAX: Input field (percentage, number)
Button to "Add Item" (to handle multiple items)



Are these form fields sufficient ??

**Reply by aayushchugh** [reply]
We'll not hardcode input fields for company details, we'll give a textarea and users can type anything there because international users can have different payment methods, tax systems etc.

**Comment by aayushchugh** [comment]
Database schema is available in wikies, I am working this task

**Comment by yashkolambekar** [comment]
### Features suggestions

- Record Payments
Once you have issued an invoice, there should be option to record payments which are for that invoice, for example, if I issue an invoice, I should be able to record payments that I receive with regards to it, in parts (it's important) and once the recorded payments reach the amount mentioned on the invoice, it should be shown as paid and until that it should be pending or partially paid. That's how it works in invoice shelf

- Client Side Dashboard
I should be able to provide my user with a login id and password at my particular link, for instance -> example.com/user/yash, on this link, if my client logs in with the credentials that I have provided them, they should be able to view everything, the invoices, the payment receipts, in a systematic manner. with some stats like total amount paid and pending

- Versioning
Invoicen should come out as a trustable platform for both the issuer and the receiver, so some things shall be restricted, like, making edits after the invoice is sent, should be restricted, one might edit it but that will create the invoice with the same number with some comment that the invoice is edited. And the original invoice should not be deleted after the edit. Just like the code changes, get's moved, but the history is saved in github, just like that.

- Templates
well this can be done later, but needs to be taken care of from now itself, I am not sure how the code looks like for the invoice renderer but please make sure that it is divided in parts, so in future, templating can be implemented and the user is not stuck with a single invoice template. User should be allowed to choose from many templates, and if required construct their own with the given components.

This is all I can think of right now
Please refer to invoiceshelf.com, it is a great tool, I use it for my invoice stuff

**Reply by aayushchugh** [reply]
I like these features we'll definitely rollout these in future after MVP is released.

**Comment by yashkolambekar** [comment]
And @aayushchugh start doing issue based development, i would like to contribute, but I am not sure about what you are planning. It would speed up the development

**Reply by aayushchugh** [reply]
We are creating issues on the go, people are closing those issues really fast that's why you can't see any open once.

**Comment by aayushchugh** [comment]
We are starting v2 now
